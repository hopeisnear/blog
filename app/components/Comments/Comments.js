/* eslint-disable no-underscore-dangle */
import React from 'react';
import { array } from 'prop-types';
import { isEmpty } from 'lodash';
import Comment from 'components/Comment';

import './Comments.scss';

const Comments = (props) => (
  <div className="Comments">
    <h2>{props.comments.length} Comments</h2>
    <ol className="comments__list">
      {props.comments.map((comment, commentIndex) => (
        <li className="list__item" key={comment._key}>
          <Comment comment={comment} index={`${commentIndex}`} />
          {!isEmpty(comment.replies) && renderReplyComments(comment.replies, commentIndex)}
        </li>
      ))}
    </ol>
  </div>
);

function renderReplyComments(replies, commentIndex) {
  return (
    <ol className="replies__list">
      {replies.map((reply, replyIndex) => (
        <li className="list__item" key={reply._key}>
          <Comment comment={reply} index={`${commentIndex}-${replyIndex}`} />
          {!isEmpty(reply.replies) && renderReplyComments(reply.replies, `${commentIndex}-${replyIndex}`)}
        </li>
      ))}
    </ol>
  );
}

Comments.propTypes = {
  comments: array.isRequired
};

export default Comments;
