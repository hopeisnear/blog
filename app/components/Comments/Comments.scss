.comments {
  margin-bottom: 60px;
  ol {
    list-style: none;
    padding-left: 0;
    li {
      margin-bottom: 20px;
      .replies {
        padding-left: 40px;
      }
    }

  }
}
