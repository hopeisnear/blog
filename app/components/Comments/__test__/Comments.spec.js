import React from 'react';
import { shallow } from 'enzyme';

import Comments from '../Comments';

describe('<Comments />', () => {
  test('should render <Comments />', () => {

    const commentsComponent = shallow(<Comments comments={createComments()} />);

    expect(commentsComponent).toMatchSnapshot();
  });

  // test('should expand add comment form when reply button is clicked', () => {
  //   const commentComponent = shallow(<Comment comment={createComment()} />);
  //
  //   commentComponent.find('button.comment__reply').simulate('click');
  //
  //   expect(commentComponent.find(AddComment).exists()).toBe(true);
  // });

  function createComments() {
    return [{ _key: '1', replies: [{ _key: '2' }, { _key: '3', replies: [{ _key: '4' }] }] }];
  }
});
