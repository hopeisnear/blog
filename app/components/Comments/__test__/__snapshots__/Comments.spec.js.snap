// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> should render <Comments /> 1`] = `
<div
  className="Comments"
>
  <h2>
    1
    Comments
  </h2>
  <ol
    className="comments__list"
  >
    <li
      className="list__item"
      key="1"
    >
      <Comment
        comment={
          Object {
            "_key": "1",
            "replies": Array [
              Object {
                "_key": "2",
              },
              Object {
                "_key": "3",
                "replies": Array [
                  Object {
                    "_key": "4",
                  },
                ],
              },
            ],
          }
        }
      />
      <ol
        className="replies__list"
      >
        <li
          className="list__item"
          key="2"
        >
          <Comment
            comment={
              Object {
                "_key": "2",
              }
            }
          />
        </li>
        <li
          className="list__item"
          key="3"
        >
          <Comment
            comment={
              Object {
                "_key": "3",
                "replies": Array [
                  Object {
                    "_key": "4",
                  },
                ],
              }
            }
          />
          <ol
            className="replies__list"
          >
            <li
              className="list__item"
              key="4"
            >
              <Comment
                comment={
                  Object {
                    "_key": "4",
                  }
                }
              />
            </li>
          </ol>
        </li>
      </ol>
    </li>
  </ol>
</div>
`;
